### ISSUE TITLE
How can I increase precision

### ISSUE THREAD

**Issue by wangtianlong1994** [ultralytics/yolov5#411]
## ❔Question
I use a custom dataset. The training dataset is about 5000 images, and the test dataset is about 500. Using your yolov3, precision and recall can reach 0.9, while using your yolov5s, precision can only reach 0.7.

## Additional context
This is yolov3:
![results](https://user-images.githubusercontent.com/48201258/87491245-38d4c780-c67a-11ea-9122-0e10b610a46c.png)
This is yolov5:
![results](https://user-images.githubusercontent.com/48201258/87491265-468a4d00-c67a-11ea-95fa-ae8955fe7e03.png)

Their datasets are the same, the only difference is that the test sets are different. V3's test sets are 20% copied from the training dataset. The test set of V5 selects a new test set. Of course, I have also tried using the new V3 test dataset, and the precision will reach 0.9

By the way, thank you for yolov5. It's really fast.

**Comment by glenn-jocher** [comment]
Its all the same, your results are identical. Your dataset is too easy to train. 

In any case if you want to use v3, train it in the v5 repo for best performance.
python train.py —cfg yolov3-spp.yaml

**Comment by wangtianlong1994** [comment]
> 都是一样的，您的结果是相同的。您的数据集太容易训练了。
> 
> 无论如何，如果您想使用v3，请在v5存储库中对其进行培训以获得最佳性能。
> python train.py — CFG yolov3-spp.yaml

Thank you for your reply. It seems that I can only change the effect by adjusting the data set, because my data set is really very simple, it is the defect detection of agricultural products, fixed background, fixed position, only the defect changes of objects. I don't need to turn on image enhancement, do I? Because my objects are the same size and aligned, they don't change. I think if image enhancement is turned on, the color and shape of the object will be changed, which will cause wrong recognition.

This is my dataset：
![2020052015280](https://user-images.githubusercontent.com/48201258/87500346-cc17f800-c68e-11ea-8bce-533df58b5f42.jpg)


Is image enhancement turned off like this?

train.py
`    # Trainloader
    dataloader, dataset = create_dataloader(train_path, imgsz, batch_size, gs, opt,
                                            hyp=hyp, augment=True, cache=opt.cache_images, rect=opt.rect)`
set:
` augment=False`

**Comment by glenn-jocher** [comment]
@wangtianlong1994 sure. You can also modify the hyperparameters as you see fit, including the augmentation ones.

**Comment by wangtianlong1994** [comment]
> @ wangtianlong1994当然。您还可以根据需要修改超参数，包括扩展参数。

Thank you for your reply

**Comment by glenn-jocher** [comment]
@wangtianlong1994 yeah no problem. hyperparams are the top of train.py.

**Comment by jingenyan** [comment]
@wangtianlong1994 itrain yolov5 with my own dataset but the map is lower than 0.1 ，can you share your dataset with me thanks！

**Comment by wangtianlong1994** [comment]
> itrain yolov5 with my own dataset but the map is lower than 0.1 ，can you share your dataset with me thanks！

sorry,I can't share my dataset because it's not me, it's my company. I think that training in yolov5, as long as the annotation is correct and the data set is simple, any data set can achieve good results

**Comment by jingenyan** [comment]
@wangtianlong1994 thanks, i 'll make more try

**Comment by saktheeswaranswan** [comment]
> > itrain yolov5 with my own dataset but the map is lower than 0.1 ，can you share your dataset with me thanks！
> 
> sorry,I can't share my dataset because it's not me, it's my company. I think that training in yolov5, as long as the annotation is correct and the data set is simple, any data set can achieve good results

this is an feedback video without any data preprocessing selection of confidence in F1 score to increase precision in some cases its desired to increase recall or maximise true positives but some times to gainable or not to use feasible gain in data precision is prepared if iam not wrong the confidence threshold which is evaluated in eval . py script between .5 to .9 increases the precision this is not hyper parameters tuning but as accuracy will not tell correct results all time we may use this --conf-tresh some time to increase when precision is needed I think we can use this conf in inference which preserves  precision which inferred or tweaked for performance but F1 score relying on an impulsive curve between 0 to 1 so this would be a feed back video continuous testing will be done to shift F1 score curve in order to trade between high precision or recall this may be done by tuning the whole hyperparameters or the whole script and need to repeat the process again because the recommended confidence is now changed whic may vary the results its better to start from scratch if some one has idea on it please tell and this will be maintained as a feedback video to

 summarise shortly the recommended value F1 score core to  confidence or the trained value of precision P and recall R  which may not suit in other conditions where we are in need to increase precision but increase the recall is another story so believing the value return by eval.py and taking the confidence threshold by tweaking conf-tresh may have bad results I guess if understand this please educate me

The feedback video is given below I made a video on this so every one to be educated or is that I have to shift F1 to confidence score curve to change everything please educate me thanks in advance 

https://youtu.be/0kONFn-28Uk

**Comment by glenn-jocher** [comment]
@saktheeswaranswan thanks for sharing your feedback and video. The YOLOv5 repository primarily focuses on the training process, and while confidence thresholds can be helpful for precision, the insights you provided on F1 scores and confidence thresholds offer valuable considerations for practitioners implementing YOLOv5 and looking to optimize precision and recall. However, for further guidance, I recommend referring to the YOLOv5 documentation or engaging with the community for additional insights. Your willingness to share your knowledge is appreciated and can help improve the understanding and usage of YOLOv5.
